File: parking-domain/src/main/java/com/chatchatabc/parking/infra/service/KVService.java
package com.chatchatabc.parking.infra.service;

public interface KVService {

    /**
     * Set key value pair
     *
     * @param key   the key
     * @param value the value
     * @param ttl   the time to live in seconds
     */
    void set(String key, String value, Long ttl);

    /**
     * Get value by key
     *
     * @param key the key
     * @return the value
     */
    String get(String key);

    /**
     * Delete value by key
     *
     * @param key the key
     */
    void delete(String key);
}

File: parking-api/src/main/kotlin/com/chatchatabc/parking/api/application/service/KVService.kt
package com.chatchatabc.parking.api.application.service

interface KVService {

    /**
     * Set key value pair
     */
    fun set(key: String, value: String, ttl: Long?)

    /**
     * Get value by key
     */
    fun get(key: String): String?

    /**
     * Delete key value pair
     */
    fun delete(key: String)
}